x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/GetGlobalCheckpointsShardAction.java

```diff
@@ -0,0 +1,242 @@
+import java.util.concurrent.TimeoutException;
+
+import static org.elasticsearch.index.seqno.SequenceNumbers.UNASSIGNED_SEQ_NO;
+
+public class GetGlobalCheckpointsShardAction extends ActionType<GetGlobalCheckpointsShardAction.Response> {
+
+    public static final GetGlobalCheckpointsShardAction INSTANCE = new GetGlobalCheckpointsShardAction();
+    public static final String NAME = "indices:monitor/fleet/global_checkpoints[s]";
+
+    private GetGlobalCheckpointsShardAction() {
+        super(NAME, GetGlobalCheckpointsShardAction.Response::new);
+    }
+
+    public static class Response extends ActionResponse {
+
+        private final long globalCheckpoint;
+
+        public Response(long globalCheckpoint) {
+            this.globalCheckpoint = globalCheckpoint;
+        }
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            globalCheckpoint = in.readLong();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeLong(globalCheckpoint);
+        }
+
+        public long getGlobalCheckpoint() {
+            return globalCheckpoint;
+        }
+    }
+
+    public static class Request extends SingleShardRequest<Request> {
+
+        private final ShardId shardId;
+        private final boolean waitForAdvance;
+        private final long currentCheckpoint;
+        private final TimeValue pollTimeout;
+
+        Request(ShardId shardId, boolean waitForAdvance, long currentCheckpoint, TimeValue pollTimeout) {
+            super(shardId.getIndexName());
+            this.shardId = shardId;
+            this.waitForAdvance = waitForAdvance;
+            this.currentCheckpoint = currentCheckpoint;
+            this.pollTimeout = pollTimeout;
+        }
+
+        Request(StreamInput in) throws IOException {
+            super(in);
+            this.shardId = new ShardId(in);
+            this.waitForAdvance = in.readBoolean();
+            this.currentCheckpoint = in.readLong();
+            this.pollTimeout = in.readTimeValue();
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        public ShardId getShardId() {
+            return shardId;
+        }
+
+        public TimeValue pollTimeout() {
+            return pollTimeout;
+        }
+
+        public boolean waitForAdvance() {
+            return waitForAdvance;
+        }
+
+        public long currentCheckpoint() {
+            return currentCheckpoint;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            shardId.writeTo(out);
+            out.writeBoolean(waitForAdvance);
+            out.writeLong(currentCheckpoint);
+            out.writeTimeValue(pollTimeout);
+        }
+    }
+
+    public static class TransportGetGlobalCheckpointAction extends TransportSingleShardAction<Request, Response> {
```

Comment:
I think we should rename the transport action to TransportGetGlobalCheckpointsShardAction to be consistent.

Could also be just `TransportAction` like in `ShardChangesAction`.